dialog.go

```diff
@@ -0,0 +1,105 @@
+	Dialog     Dialog  `json:"dialog"`      //Required.
+}
+
+type Dialog struct {
+	Title           string           `json:"title"`                         //Required.
+	CallbackId      string           `json:"callback_id"`                   //Required.
+	Elements        []DialogElement  `json:"elements"`                      //Required.
+	SubmitLabel     string           `json:"submit_label,omitempty"`        //Optional. Default value is 'Submit'
+	NotifyOnCancel  bool             `json:"notify_on_cancel,omitempty"`    //Optional. Default value is false
+}
+
+type DialogElement interface {}
+
+type DialogTextElement struct {
+	Label        string     `json:"label"`                  //Required.
+	Name         string     `json:"name"`                   //Required.
+	Type         string     `json:"type"`                   //Required. Allowed values: "text", "textarea", "select".
+	Placeholder  string     `json:"placeholder,omitempty"`  //Optional.
+	Optional     bool       `json:"optional,omitempty"`     //Optional. Default value is false
+	Value        string     `json:"value,omitempty"`        //Optional.
+	MaxLength    int        `json:"max_length,omitempty"`   //Optional.
+	MinLength    int        `json:"min_length,omitempty"`   //Optional,. Default value is 0
+	Hint         string     `json:"hint,omitempty"`         //Optional.
+	Subtype      string     `json:"subtype,omitempty"`      //Optional. Allowed values: "email", "number", "tel", "url".
+}
+
+type DialogSelectElement struct {
+	Label            string                 `json:"label"`                          //Required.
+	Name             string                 `json:"name"`                           //Required.
+	Type             string                 `json:"type"`                           //Required. Allowed values: "text", "textarea", "select".
+	Placeholder      string                 `json:"placeholder,omitempty"`          //Optional.
+	Optional         bool                   `json:"optional,omitempty"`             //Optional. Default value is false
+	Value            string                 `json:"value,omitempty"`                //Optional.
+	DataSource       string                 `json:"data_source,omitempty"`          //Optional. Allowed values: "users", "channels", "conversations", "external".
+	SelectedOptions  string                 `json:"selected_options,omitempty"`     //Optional. Default value for "external" only
+	Options          []DialogElementOption  `json:"options,omitempty"`              //One of options or option_groups is required.
+	OptionGroups     []DialogElementOption  `json:"option_groups,omitempty"`        //Provide up to 100 options.
+}
+
+type DialogElementOption struct {
+	Label  string   `json:"label"`	//Required.
+	Value  string   `json:"value"`	//Required.
+}
+
+// DialogCallback is sent from Slack when a user submits a form from within a dialog
+type DialogCallback struct {
+	Type         string             `json:"dialog_submission"`
```

Comment:
would this not be `json:"type"`?

Reply:
Whoops... Nice spot. I'll update it now.

Reply:
Done.